components/substitutions/index.rst

```diff
@@ -0,0 +1,181 @@
+Substitutions
+=============
+
+.. seo::
+    :description: How to use substitutions in ESPHome
+    :image: settings.svg
+
+ESPHome has a powerful way to reduce repetition in configuration files: substitutions.
+With substitutions, you can have a single generic source file for all nodes of one kind and
+substitute expressions in as required.
+
+.. code-block:: yaml
+
+    substitutions:
+      devicename: livingroom
+      upper_devicename: Livingroom
+
+    esphome:
+      name: $devicename
+      # ...
+
+    sensor:
+    - platform: dht
+      # ...
+      temperature:
+        name: ${upper_devicename} Temperature
+      humidity:
+        name: ${upper_devicename} Humidity
```

Comment:
With the `friendly_name` being added, this is not a good example to have here.

Reply:
Are we still need to add this anyway? It is already prefix it.

Reply:
Yeah that was my point why this is a bad example. This example should do something else that is more useful to users.